heat/core/tensor.py

```diff
@@ -29,17 +30,38 @@ def dtype(self):
     def gshape(self):
         return self.__gshape
 
+    @property
+    def size(self):
+        try:
+            return reduce(lambda i, j: i * j, self.array_shape)
+        except TypeError:
+            # todo: throw a soft warning about taking the number of elements of a single element tensor
+            return 1
+
+    @property
+    def gnumel(self):
+        return self.size
+
+    @property
+    def lnumel(self):
+        return reduce(lambda i, j: i * j, self.lshape)
+
     @property
     def lshape(self):
         if len(self.__array.shape) == len(self.__gshape):
             return tuple(self.__array.shape)
         # edge case when the local data tensor receives no elements after chunking
-        return self.__gshape[:self.__split] + (0,) + self.__gshape[self.split + 1:]
+        return self.__array.shape
+        # return self.__gshape[:self.__split] + (0,) + self.__gshape[self.split + 1:]
```

Comment:
please remove dead code in comments

in what way is the old implementation insufficient?

Reply:
the old way would return a 0 along the split axis and not the true local shape. 
i.e. when run with two processes
```python
z = ht.zeros([5, 4, 3], split=1)
z.lshape
>>> (5, 2, 3)
z.gshape[:z.split] + (0,) + z.gshape[z.split + 1:]
>>> (5, 0, 3)
```

I had left the old code so we could discuss this and i wouldn't have to look elsewhere for it.

Reply:
Okay, so the computation is actually more tricky than this even. Considers this:

`mpirun -np 5 python -c "import heat as ht; x = ht.zeros((5,4,3), split=1)"`

Rank 4 should correctly report that the lshape is (5,0,3), while all the others should report (5,1,3). While not always necessary sensible to have a such a high degree of parallelization with such a small amount of data, there are situations where this might happens.

Reply:
testing with the new implementation does return that if it is the case:

```python
z = ht.zeros([5, 4, 3], split=1)
print(z.lshape)
>>>mpiexec -n 5 python testing.py 
>>>(5, 1, 3)
>>>(5, 1, 3)
>>>(5, 1, 3)
>>>(5, 1, 3)
>>>(5, 0, 3)

Reply:
Alright, looks good